### PR TITLE
Add --test-env option to tests/runtest.sh to set stress mode environm…

### PR DESCRIPTION
**Pull request by sejongoh**
Add --test-env option to set environment variables just before running tests. This is similar to the TestEnv option of runtest.cmd.

### CODE REVIEW

**Comment by sejongoh** [comment]
Please don't review this yet. I need to update one thing.

**Comment by sejongoh** [comment]
@mmitche Could you please review this? This changeset adds --test-env to runtest.sh which is similar to TestEnv in the Windows script.

cc @dotnet/jit-contrib

**Comment by sejongoh** [comment]
I will change this to using __TestEnv.

File: tests/runtest.sh [hunk 1]
```diff
@@ -40,6 +40,7 @@ function print_usage {
     echo '  -v, --verbose                : Show output from each test.'
     echo '  -h|--help                    : Show usage information.'
     echo '  --useServerGC                : Enable server GC for this test run'
+    echo '  --testEnv                    : Script to set environment variables for tests'
```
Comment (sejongoh) [inline tests/runtest.sh]: I will change --testEnv to --test-env.

**Comment by sejongoh** [comment]
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test  OSX x64 Checked Build and Test

**Comment by sejongoh** [comment]
@mmitche Could you take a look? I updated runtest.sh to use __TestEnv internally.

File: netci.groovy [hunk 2]
```diff
@@ -157,16 +156,39 @@ def static genStressModeScriptStep(def os, def stressModeName, def stressModeVar
     else {
         // For these we don't use a script, we use directly
         stepScript += "echo Setting variables for ${stressModeName}\n"
+        stepScript += "rm -rf ${stepScriptLocation}\n"
```
Comment (mmitche) [inline netci.groovy]: I don't think you need -r (recursive) here

File: tests/runtest.sh [hunk 3]
```diff
@@ -789,8 +794,27 @@ copy_test_native_bin_to_test_root
...
+    # Set __TestEnv that will be executed just before running tests
+    absTestEnvPath=`readlink -f ${testEnv}`
+    export __TestEnv='source '${absTestEnvPath}
+fi
+
+# Still support setting COMPlus_GCStress before runtest.sh but recommend
+# using --test-env option. 
+if [ -z "$hasGCStressEnv" ]; then
+    if [ -n "$COMPlus_GCStress" ]; then
+        hasGCStressEnv='true'
```
Comment (mmitche) [inline tests/runtest.sh]: Is there a reason we're using a string for 'true' (vs. false?) instead of booleans?

File: netci.groovy (hunk 2 above)
Comment (sejongoh) [inline netci.groovy]: Right. '-r' is unnecessary.

File: tests/runtest.sh (hunk 3 above)
Comment (sejongoh) [inline tests/runtest.sh]: I think that boolean true works here too. I just used 'true' since hasGCStressEnv can be defined as a string in line 802.

**Comment by sejongoh** [comment]
@mmitche PTAL. I changed hasGCStress variable to GCStressLevel that contains the value of stress level.

**Comment by mmitche** [comment]
Thanks, LGTM

File: tests/runtest.sh [hunk 4]
```diff
@@ -756,8 +761,27 @@ copy_test_native_bin_to_test_root
 load_unsupported_tests
 load_failing_tests
 
-if [ -n "$COMPlus_GCStress" ]; then
-    scriptPath=$(dirname $0)
+scriptPath=$(dirname $0)
+
+# Check if environment variables are provided
+if [ ! -z "$testEnv" ]; then
```
Comment (RussKeldorph) [inline tests/runtest.sh]: Is ! -z the same as -n?  Here and below as well.

Comment (sejongoh) [inline tests/runtest.sh]: yes. ! -z is the same as -n.

**Pull request merged**